src/controller/SetUpCodePairer.cpp

```diff
@@ -286,8 +289,21 @@ CHIP_ERROR SetUpCodePairer::StartDiscoveryOverWiFiPAF(SetupPayload & payload)
 CHIP_ERROR SetUpCodePairer::StopDiscoveryOverWiFiPAF()
 {
     mWaitingForDiscovery[kWiFiPAFTransport] = false;
+
 #if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
-    DeviceLayer::ConnectivityMgr().WiFiPAFCancelIncompleteSubscribe();
+    if (mPAFSessionId != WiFiPAF::kUndefinedWiFiPafSessionId)
+    {
+        WiFiPAF::WiFiPAFSession sessionInfo = { .role = WiFiPAF::WiFiPafRole::kWiFiPafRole_Subscriber, .id = mPAFSessionId };
+        WiFiPAF::WiFiPAFLayer & pafLayer    = WiFiPAF::WiFiPAFLayer::GetWiFiPAFLayer();
+        WiFiPAF::WiFiPAFSession * pSession  = pafLayer.GetPAFInfo(WiFiPAF::PafInfoAccess::kAccSessionId, sessionInfo);
+        if ((pSession != nullptr) && (pSession->peer_id == WiFiPAF::kUndefinedWiFiPafSessionId))
```

Comment:
Again, all this checking should be hidden under the hood in the PAF APIs, I would think.  We should tell PAF to stop discovery, it should presumably not tear down sessions with peers?  If "stop discovery" needs an identifier that the "start discovery" API returned, so we know which discovery process to stop, that seems fine; maybe that's mPAFSessionId?